src/YieldDisburser.sol

```diff
@@ -142,50 +143,16 @@ contract YieldDisburser is OwnableUpgradeable {
         return projects.length;
     }
 
-    /**
-     * @notice Return the current votes cast
-     * @return address[] Array of voters
-     * @return uint256[][] Array of vote distributions
-     */
-    function currentVotesCasted() public view returns (address[] memory, uint256[][] memory) {
-        uint256 projectsLength = projects.length;
-        uint256 votersLength = voters.length;
-        uint256[][] memory voterDistributions = new uint256[][](votersLength);
-        for (uint256 i; i < votersLength; ++i) {
-            uint256[] memory votes = new uint256[](projectsLength);
-            address voter = voters[i];
-            uint256 voterPower = this.getVotingPowerForPeriod(lastClaimedBlockNumber, Time.blockNumber(), voter);
-            uint256[] memory voterDistribution = holderToDistribution[voter];
-            uint256 vote;
-            for (uint256 j; j < projectsLength; ++j) {
-                vote = ((voterPower * voterDistribution[j] * PRECISION) / holderToDistributionTotal[voter]) / PRECISION;
-                votes[j] = vote;
-            }
-            voterDistributions[i] = votes;
-        }
-        return (voters, voterDistributions);
-    }
-
-    /**
-     * @notice Return the current votes cast by a specified holder
-     * @param _holder Holder to return the current votes cast of
-     * @return uint256[] Distribution of votes cast by the specified holder
-     */
-    function currentVoteCast(address _holder) public view returns (uint256[] memory) {
-        uint256[] memory vote = holderToDistribution[_holder];
-        return vote;
-    }
-
     /**
      * @notice Determine if the yield distribution is available
      * @return bool Flag indicating if the yield distribution is able to be claimed
      * @return bytes Function selector used to distribute the yield
      */
     function resolveYieldDistribution() public view returns (bool, bytes memory) {
-        uint48 _now = Time.timestamp();
+        if (currentVotes == 0) revert NoVotesCasted();
         uint256 balance = (BREAD.balanceOf(address(this)) + BREAD.yieldAccrued());
         if (balance < projects.length) revert YieldTooLow(balance);
-        if (_now < lastClaimedTimestamp + minTimeBetweenClaims) {
+        if (block.number < lastClaimedTimestamp + cycleLength) {
```

Comment:
`block.number` is not the same as `block.timestamp` .. this will result in the yield always being in the "already claimed" state after it has been claimed the first time.

Reply:
addressed by 9e6a03a and others